Pocket/pocket-ios
# In 1482 - App Badge Not Displaying

## Summary
Changes how we save the AppBadge flag to absolute value rather than just flipping the existing value. For some reason the value coming out of UserDefaults was always true, which would then be toggled to false.
Also add some analytics to appBadge events (toggling the setting and one for rejected permissions.)

## References 
https://getpocket.atlassian.net/browse/IN-1482

## Implementation Details
Passing the actual value of the toggle to UserDefaults rather than inverting the existing value.

## Test Steps
You should now be able to enable and disable the app badge from Pocket Settings.

## PR Checklist:

## Screenshots
![appBadge](https://github.com/Pocket/pocket-ios/assets/8590135/1d4113c7-08dc-40be-a8ae-1cf6f2caa582)
